language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1277,6 +1276,44 @@ public static TypeSymbol getRawType(TypeSymbol typeDescriptor) {
         return typeDescriptor;
     }
 
+    /**
+     * Check if the provided union type is a union of members of provided type desc kind.
+     *
+     * @param typeSymbol Union type symbol
+     * @param typeDescKind    Type desc kind
+     * @return true if provided union contains members of provided type desc kind
+     */
+    public static boolean isUnionOfType(TypeSymbol typeSymbol, TypeDescKind typeDescKind) {
+        return typeSymbol.typeKind() == TypeDescKind.UNION &&
+                ((UnionTypeSymbol) typeSymbol).memberTypeDescriptors().stream()
+                        .map(CommonUtil::getRawType)
+                        .map(TypeSymbol::typeKind)
+                        .allMatch(kind -> kind == typeDescKind);
+    }
+
+    /**
+     * Given a type symbol, this method will get the record type symbols in the provided type. i.e. the provided type
+     * can be a record or a union of records for this to work.
+     *
+     * @param typeSymbol Type symbol from which record types need to be extracted.
+     * @return List of record type symbols, wrapped with a raw type and broader type container.
+     */
+    public static List<RawTypeSymbolWrapper<RecordTypeSymbol>> getRecordTypeSymbols(TypeSymbol typeSymbol) {
+        TypeSymbol rawType = CommonUtil.getRawType(typeSymbol);
+        if (rawType.typeKind() == TypeDescKind.RECORD) {
+            return Collections.singletonList(RawTypeSymbolWrapper.from(typeSymbol, (RecordTypeSymbol) rawType));
+        } else if (rawType.typeKind() == TypeDescKind.UNION) {
```

Comment:
Since we return, we don't need `else-if`. Let's stick to a set of `if` clauses for better readability